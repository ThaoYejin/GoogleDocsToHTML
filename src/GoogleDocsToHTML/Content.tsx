import { docs_v1 } from "googleapis";
import React from "react";
import { Heading } from "./Heading";
import { Paragraph } from "./Paragraph";
import { Table } from "./Table";
export interface ContentProps {
    doc: docs_v1.Schema$Document;
    body? : docs_v1.Schema$StructuralElement[];
}

export const Content: React.FC<ContentProps> = ({
    doc,
    body
}) => {
    var content: React.ReactNode[];
    content = [];
    body?.forEach(function (value, index) {
        var pObj = value.paragraph;

        //Display Heading
        if (pObj?.paragraphStyle?.namedStyleType?.indexOf("HEADING") != -1) {
            content?.push(
                <Heading key={index} paraObj={pObj}></Heading>
            )
        }

        //Display paragraph
        if (pObj?.paragraphStyle?.namedStyleType?.indexOf("HEADING") == -1 && value.table == undefined && pObj.bullet == undefined) {
            content?.push(<Paragraph key={index} paraObj={pObj} doc={doc}></Paragraph>)
        }

        //Display table
        if(value.table != undefined){
            content.push(<Table doc={doc} table={value.table}></Table>)
        }
    })
    return (
        <>
            {content}
        </>
    )
}