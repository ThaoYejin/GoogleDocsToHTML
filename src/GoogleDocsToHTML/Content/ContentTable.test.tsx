import React from 'react';
import renderer from 'react-test-renderer';
import { Content } from './Content';
import * as doc from './ContentTableTest.json';

it('renders correctly', () => {
  var body = [
    {
      "endIndex": 1,
      "sectionBreak": {
        "sectionStyle": {
          "columnSeparatorStyle": "NONE",
          "contentDirection": "LEFT_TO_RIGHT",
          "sectionType": "CONTINUOUS"
        }
      }
    },
    {
      "startIndex": 1,
      "endIndex": 2,
      "paragraph": {
        "elements": [
          {
            "startIndex": 1,
            "endIndex": 2,
            "textRun": {
              "content": "\n",
              "textStyle": {
                "foregroundColor": {
                  "color": {
                    "rgbColor": {
                      "red": 0.2509804,
                      "green": 0.2509804,
                      "blue": 0.2509804
                    }
                  }
                },
                "fontSize": {
                  "magnitude": 12,
                  "unit": "PT"
                }
              }
            }
          }
        ],
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT"
        }
      }
    },
    {
      "startIndex": 2,
      "endIndex": 36,
      "table": {
        "rows": 2,
        "columns": 2,
        "tableRows": [
          {
            "startIndex": 3,
            "endIndex": 19,
            "tableCells": [
              {
                "startIndex": 4,
                "endIndex": 11,
                "content": [
                  {
                    "startIndex": 5,
                    "endIndex": 11,
                    "paragraph": {
                      "elements": [
                        {
                          "startIndex": 5,
                          "endIndex": 11,
                          "textRun": {
                            "content": "First\n",
                            "textStyle": {
                              "foregroundColor": {
                                "color": {
                                  "rgbColor": {
                                    "red": 0.2509804,
                                    "green": 0.2509804,
                                    "blue": 0.2509804
                                  }
                                }
                              },
                              "fontSize": {
                                "magnitude": 12,
                                "unit": "PT"
                              }
                            }
                          }
                        }
                      ],
                      "paragraphStyle": {
                        "namedStyleType": "NORMAL_TEXT",
                        "lineSpacing": 100,
                        "direction": "LEFT_TO_RIGHT",
                        "avoidWidowAndOrphan": false
                      }
                    }
                  }
                ],
                "tableCellStyle": {
                  "rowSpan": 1,
                  "columnSpan": 1,
                  "backgroundColor": {},
                  "paddingLeft": {
                    "magnitude": 5,
                    "unit": "PT"
                  },
                  "paddingRight": {
                    "magnitude": 5,
                    "unit": "PT"
                  },
                  "paddingTop": {
                    "magnitude": 5,
                    "unit": "PT"
                  },
                  "paddingBottom": {
                    "magnitude": 5,
                    "unit": "PT"
                  },
                  "contentAlignment": "TOP"
                }
              },
              {
                "startIndex": 11,
                "endIndex": 19,
                "content": [
                  {
                    "startIndex": 12,
                    "endIndex": 19,
                    "paragraph": {
                      "elements": [
                        {
                          "startIndex": 12,
                          "endIndex": 19,
                          "textRun": {
                            "content": "Second\n",
                            "textStyle": {
                              "foregroundColor": {
                                "color": {
                                  "rgbColor": {
                                    "red": 0.2509804,
                                    "green": 0.2509804,
                                    "blue": 0.2509804
                                  }
                                }
                              },
                              "fontSize": {
                                "magnitude": 12,
                                "unit": "PT"
                              }
                            }
                          }
                        }
                      ],
                      "paragraphStyle": {
                        "namedStyleType": "NORMAL_TEXT",
                        "lineSpacing": 100,
                        "direction": "LEFT_TO_RIGHT",
                        "avoidWidowAndOrphan": false
                      }
                    }
                  }
                ],
                "tableCellStyle": {
                  "rowSpan": 1,
                  "columnSpan": 1,
                  "backgroundColor": {},
                  "paddingLeft": {
                    "magnitude": 5,
                    "unit": "PT"
                  },
                  "paddingRight": {
                    "magnitude": 5,
                    "unit": "PT"
                  },
                  "paddingTop": {
                    "magnitude": 5,
                    "unit": "PT"
                  },
                  "paddingBottom": {
                    "magnitude": 5,
                    "unit": "PT"
                  },
                  "contentAlignment": "TOP"
                }
              }
            ],
            "tableRowStyle": {
              "minRowHeight": {
                "unit": "PT"
              }
            }
          },
          {
            "startIndex": 19,
            "endIndex": 35,
            "tableCells": [
              {
                "startIndex": 20,
                "endIndex": 27,
                "content": [
                  {
                    "startIndex": 21,
                    "endIndex": 27,
                    "paragraph": {
                      "elements": [
                        {
                          "startIndex": 21,
                          "endIndex": 27,
                          "textRun": {
                            "content": "Third\n",
                            "textStyle": {
                              "foregroundColor": {
                                "color": {
                                  "rgbColor": {
                                    "red": 0.2509804,
                                    "green": 0.2509804,
                                    "blue": 0.2509804
                                  }
                                }
                              },
                              "fontSize": {
                                "magnitude": 12,
                                "unit": "PT"
                              }
                            }
                          }
                        }
                      ],
                      "paragraphStyle": {
                        "namedStyleType": "NORMAL_TEXT",
                        "lineSpacing": 100,
                        "direction": "LEFT_TO_RIGHT",
                        "avoidWidowAndOrphan": false
                      }
                    }
                  }
                ],
                "tableCellStyle": {
                  "rowSpan": 1,
                  "columnSpan": 1,
                  "backgroundColor": {},
                  "paddingLeft": {
                    "magnitude": 5,
                    "unit": "PT"
                  },
                  "paddingRight": {
                    "magnitude": 5,
                    "unit": "PT"
                  },
                  "paddingTop": {
                    "magnitude": 5,
                    "unit": "PT"
                  },
                  "paddingBottom": {
                    "magnitude": 5,
                    "unit": "PT"
                  },
                  "contentAlignment": "TOP"
                }
              },
              {
                "startIndex": 27,
                "endIndex": 35,
                "content": [
                  {
                    "startIndex": 28,
                    "endIndex": 35,
                    "paragraph": {
                      "elements": [
                        {
                          "startIndex": 28,
                          "endIndex": 35,
                          "textRun": {
                            "content": "Fourth\n",
                            "textStyle": {
                              "foregroundColor": {
                                "color": {
                                  "rgbColor": {
                                    "red": 0.2509804,
                                    "green": 0.2509804,
                                    "blue": 0.2509804
                                  }
                                }
                              },
                              "fontSize": {
                                "magnitude": 12,
                                "unit": "PT"
                              }
                            }
                          }
                        }
                      ],
                      "paragraphStyle": {
                        "namedStyleType": "NORMAL_TEXT",
                        "lineSpacing": 100,
                        "direction": "LEFT_TO_RIGHT",
                        "avoidWidowAndOrphan": false
                      }
                    }
                  }
                ],
                "tableCellStyle": {
                  "rowSpan": 1,
                  "columnSpan": 1,
                  "backgroundColor": {},
                  "paddingLeft": {
                    "magnitude": 5,
                    "unit": "PT"
                  },
                  "paddingRight": {
                    "magnitude": 5,
                    "unit": "PT"
                  },
                  "paddingTop": {
                    "magnitude": 5,
                    "unit": "PT"
                  },
                  "paddingBottom": {
                    "magnitude": 5,
                    "unit": "PT"
                  },
                  "contentAlignment": "TOP"
                }
              }
            ],
            "tableRowStyle": {
              "minRowHeight": {
                "unit": "PT"
              }
            }
          }
        ],
        "tableStyle": {
          "tableColumnProperties": [
            {
              "widthType": "EVENLY_DISTRIBUTED"
            },
            {
              "widthType": "EVENLY_DISTRIBUTED"
            }
          ]
        }
      }
    },
    {
      "startIndex": 36,
      "endIndex": 37,
      "paragraph": {
        "elements": [
          {
            "startIndex": 36,
            "endIndex": 37,
            "textRun": {
              "content": "\n",
              "textStyle": {
                "foregroundColor": {
                  "color": {
                    "rgbColor": {
                      "red": 0.2509804,
                      "green": 0.2509804,
                      "blue": 0.2509804
                    }
                  }
                },
                "fontSize": {
                  "magnitude": 12,
                  "unit": "PT"
                }
              }
            }
          }
        ],
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT"
        }
      }
    },
    {
      "startIndex": 37,
      "endIndex": 38,
      "paragraph": {
        "elements": [
          {
            "startIndex": 37,
            "endIndex": 38,
            "textRun": {
              "content": "\n",
              "textStyle": {
                "foregroundColor": {
                  "color": {
                    "rgbColor": {
                      "red": 0.2509804,
                      "green": 0.2509804,
                      "blue": 0.2509804
                    }
                  }
                },
                "fontSize": {
                  "magnitude": 12,
                  "unit": "PT"
                }
              }
            }
          }
        ],
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT"
        }
      }
    }
  ];

    const tree = renderer
        .create(<Content doc={doc} body={body} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
})