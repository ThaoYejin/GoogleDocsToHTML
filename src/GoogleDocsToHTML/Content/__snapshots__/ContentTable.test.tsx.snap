// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h6 />,
  <p>
    <span
      style={
        Object {
          "color": "rgb(64.000002,64.000002,64.000002)",
          "fontSize": 12,
        }
      }
    >
      

    </span>
  </p>,
  <h6 />,
  <table
    style={
      Object {
        "borderCollapse": "collapse",
      }
    }
  >
    <tr>
      <td
        colSpan={1}
        rowSpan={1}
        style={
          Object {
            "borderCollapse": "collapse",
          }
        }
      >
        <p>
          <span
            style={
              Object {
                "color": "rgb(64.000002,64.000002,64.000002)",
                "fontSize": 12,
              }
            }
          >
            First

          </span>
        </p>
      </td>
      <td
        colSpan={1}
        rowSpan={1}
        style={
          Object {
            "borderCollapse": "collapse",
          }
        }
      >
        <p>
          <span
            style={
              Object {
                "color": "rgb(64.000002,64.000002,64.000002)",
                "fontSize": 12,
              }
            }
          >
            Second

          </span>
        </p>
      </td>
    </tr>
    <tr>
      <td
        colSpan={1}
        rowSpan={1}
        style={
          Object {
            "borderCollapse": "collapse",
          }
        }
      >
        <p>
          <span
            style={
              Object {
                "color": "rgb(64.000002,64.000002,64.000002)",
                "fontSize": 12,
              }
            }
          >
            Third

          </span>
        </p>
      </td>
      <td
        colSpan={1}
        rowSpan={1}
        style={
          Object {
            "borderCollapse": "collapse",
          }
        }
      >
        <p>
          <span
            style={
              Object {
                "color": "rgb(64.000002,64.000002,64.000002)",
                "fontSize": 12,
              }
            }
          >
            Fourth

          </span>
        </p>
      </td>
    </tr>
  </table>,
  <p>
    <span
      style={
        Object {
          "color": "rgb(64.000002,64.000002,64.000002)",
          "fontSize": 12,
        }
      }
    >
      

    </span>
  </p>,
  <p>
    <span
      style={
        Object {
          "color": "rgb(64.000002,64.000002,64.000002)",
          "fontSize": 12,
        }
      }
    >
      

    </span>
  </p>,
]
`;
